host-interaction/process/inject/allocate-rw-memory.yml

```diff
@@ -0,0 +1,26 @@
+rule:
+  meta:
+    name: allocate RW memory
+    namespace: host-interaction/process/inject
+    author: 0x534a@mailbox.org
+    scope: basic block
+    att&ck:
+      - Defense Evasion::Process Injection [T1055]
+    mbc:
+      - Memory::Allocate Memory [C0007]
+    examples:
+      - Practical Malware Analysis Lab 17-02.dll_:0x1000D10D
+  features:
+    - and:
+      - or:
+        - api: kernel32.VirtualAlloc
+        - api: kernel32.VirtualAllocEx
+        - api: kernel32.VirtualAllocExNuma
+        - api: kernel32.VirtualProtect
+        - api: kernel32.VirtualProtectEx
+        - api: ntdll.NtAllocateVirtualMemory
+        - api: ntdll.ZwAllocateVirtualMemory
+        - api: ntdll.NtMapViewOfSection
+        - api: ntdll.ZwMapViewOfSection
```

Comment:
could move these to a library rule `allocate memory` and use a match in the RW and RWX rules

Reply:
Fixed in aa94a33 but there is a linter error. See comment below.